evap/staff/forms.py

```diff
@@ -233,27 +255,94 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-
-        self.fields["responsibles"].queryset = UserProfile.objects.exclude(is_active=False)
-        if self.instance.pk:
-            self.fields["responsibles"].queryset |= UserProfile.objects.filter(
-                pk__in=[user.pk for user in self.instance.responsibles.all()]
-            )
-
+        self._set_responsibles_queryset()
         if not self.instance.can_be_edited_by_manager:
             disable_all_fields(self)
 
-    def validate_unique(self):
-        super().validate_unique()
-        # name_xy and semester are unique together. This will be treated as a non-field-error since two
-        # fields are involved. Since we only show the name_xy field to the user, assign that error to this
-        # field. This hack is not documented, so it might be broken when you are reading this.
-        for e in self.non_field_errors().as_data():
-            if e.code == "unique_together" and "unique_check" in e.params:
-                if "semester" in e.params["unique_check"]:
-                    # The order of the fields is probably determined by the unique_together constraints in the Course class.
-                    name_field = e.params["unique_check"][1]
-                    self.add_error(name_field, e)
+
+class CourseCopyForm(CourseFormMixin, forms.ModelForm):
+    semester = forms.ModelChoiceField(Semester.objects.all())
+    vote_start_datetime = forms.DateTimeField(label=_("Start of associated evaluations"), localize=True)
+    vote_end_date = forms.DateField(label=_("End of associated evaluations"), localize=True)
+
+    class Meta(CourseForm.Meta):
+        pass
+
+    field_order = ["semester"]
+
+    def __init__(self, data=None, instance: Course = None):
+        self.old_course = instance
+        opts = self._meta
+        initial = forms.models.model_to_dict(instance, opts.fields, opts.exclude)
+        super().__init__(data=data, initial=initial)
+        self._set_responsibles_queryset()
+
+    # To ensure we don't forget about copying a relevant field, we explicitly list copied and ignored fields and test against that
+    EVALUATION_COPIED_FIELDS = {
+        "name_de",
+        "name_en",
+        "weight",
+        "is_single_result",
+        "is_rewarded",
+        "is_midterm_evaluation",
+        "allow_editors_to_edit",
+        "wait_for_grade_upload_before_publishing",
+    }
+
+    EVALUATION_EXCLUDED_FIELDS = {
+        "id",
+        "course",
+        "can_publish_text_reuslts",
```

Comment:
```suggestion

```
i first wondered how the typo in here (`reuslts`) did not make the test fail but then saw the attribute is listed below again